clang/test/Sema/warn-lifetime-analysis-nocfg.cpp

```diff
@@ -793,3 +793,108 @@ void test13() {
 }
 
 } // namespace GH100526
+
+namespace lifetime_capture_by {
+struct S {
+  const int *x;
+  void captureInt(const int&x [[clang::lifetime_capture_by(this)]]) { this->x = &x; }
+  void captureSV(std::string_view sv [[clang::lifetime_capture_by(this)]]);
+};
+///////////////////////////
+// Detect dangling cases.
+///////////////////////////
+void captureInt(const int&x [[clang::lifetime_capture_by(s)]], S&s);
+void captureRValInt(int&&x [[clang::lifetime_capture_by(s)]], S&s);
+void noCaptureInt(int x [[clang::lifetime_capture_by(s)]], S&s);
```

Comment:
Should we diagnose this as a misuse of the attribute?

Reply:
+1.

Reply:
SG. I will add an error diagnositc on use of this annotation on non-reference and non-view types. But I prefer to do this in a followup PR.

Reply:
Since this diagnostic is not really lifetime analysis related, I think doing this in a separate PR is the cleaner way to go.